### ISSUE TITLE
Landlab install failed

### ISSUE THREAD

**Issue by KevRozmiarek** [landlab/landlab#1406]
Hi all,

I am attempting to install Landlab and have hit a snag with the install line.

$ conda install landlab -c conda-forge

The error message I have received is attached.

![Screenshot from 2022-04-03 18-49-38](https://user-images.githubusercontent.com/22139518/161617152-bab939d9-0e81-420a-a531-c0deed74c09f.png)

OS: Pop!_OS 21.10
Conda: 2.1.2
Python: 3.8

Thanks.

**Comment by mcflugen** [comment]
@KevRozmiarek This is an error I don't I've seen before.

It looks like you are wanting to install *landlab* so that you can run the notebooks, is that right? It also looks like you might be following some outdated instructions. Could you point me to the instructions you are following, or the steps you took to get to this place?

This following commands will install the latest version of *landlab* (v2.4.1) into a new environment for running the notebooks, download the latest version of the notebooks, and then install requirements for running the notebooks.
```bash
$ conda create -n landlab-notebooks
$ conda activate landlab-notebooks
$ conda install landlab -c conda-forge
$ curl -O -L https://github.com/landlab/landlab/archive/refs/tags/v2.4.1.tar.gz
$ tar xvfz v2.4.1.tar.gz
$ cd landlab-2.4.1
$ conda install --file=requirements-notebooks.txt
```
You should now be able to run the notebooks under the *notebooks* folder. Is this more or less what you did?

**Comment by KevRozmiarek** [comment]
Hi @mcflugen I aspire to produce errors not seen before. Also sorry for the late response.

You are correct about installing landlab for notebooks. I was using the instructions found here. https://landlab.readthedocs.io/en/master/install/index.html

None of the instructions worked for me. I was able to work with @BCampforts to resolve my issue. Here is what I did:

Clone landlab to local
```
$ conda create -n ldev
$ conda activate ldev
$ conda install -c conda-forge -c defaults --file=requirements.txt --file=requirements-dev.txt --file=requirements-notebooks.txt --file=requirements-testing.txt
$ pip install  -e .
```

Kevin

**Comment by mcflugen** [comment]
@KevRozmiarek Great to hear you got things working! 🎉

@BCampforts had you do a developer install of *landlab* but I would still like to make sure our installation instructions are correct. If it's not too much trouble, would you mind helping me out with that?

To install the latest version of *landlab*, you should do the following
```bash
$ conda create -n landlab-notebooks
$ conda activate landlab-notebooks
$ conda install landlab -c conda-forge
```
You say that didn't work for you though, is that right? What was the error you received?

**Comment by BCampforts** [comment]
Yeah, we had to use to the dev install to get access to the PriorityFloodFlowRouter. I think the error of @KevRozmiarek  got (on a Linux machine) was an ImportError looking like this using the default installation (?): 
```ImportError: cannot import name "PriorityFloodFlowRouter" from 'landlab.components' (C:\Users\user\anaconda3\lib\site-packages\landlab\components\__init__.py```
This is the error @josiearcuri got on a windows machine

**Comment by mcflugen** [comment]
@BCampforts Got it. Thank you.

That makes sense. It wasn't really an install issue at all. It sounds like we should update our instructions for running the *landlab* notebooks. If one wants to run the notebooks with a particular *landlab* release, it is important that they match the version of the notebooks with the version of *landlab*. Since you were wanting to run a pre-release version of the notebooks, you also need to install a pre-release version of *landlab*.

The instructions should then be something along the lines of,
1. To run the latest release of the notebooks,
```bash
$ conda create -n landlab-notebooks
$ conda activate landlab-notebooks
$ conda install landlab=2.4.1 -c conda-forge
$ curl -O -L https://github.com/landlab/landlab/archive/refs/tags/v2.4.1.tar.gz
$ tar xvfz v2.4.1.tar.gz
$ cd landlab-2.4.1
$ conda install --file=requirements-notebooks.txt
```
2. To run a development version of the notebooks,
```bash
$ conda create -n landlab-notebooks
$ conda activate landlab-notebooks
$ curl -O -L https://github.com/landlab/landlab/archive/refs/heads/master.tar.gz
$ tar xvfz master.tar.gz
$ cd landlab-master
$ conda install --file=requirements.txt
$ pip install.
$ conda install --file=requirements-notebooks.txt
```
